pages/lists/[listId]/contributors.tsx

```diff
@@ -1,9 +1,110 @@
+import { GetServerSidePropsContext } from "next";
+import { createPagesServerClient } from "@supabase/auth-helpers-nextjs";
 import ListPageLayout from "layouts/lists";
+import ContributorTable from "components/organisms/ContributorsTable/contributors-table";
+import PaginationResults from "components/molecules/PaginationResults/pagination-result";
+import ContributorListTableHeaders from "components/molecules/ContributorListTableHeader/contributor-list-table-header";
+import Pagination from "components/molecules/Pagination/pagination";
+import { fetchApiData, validateListPath } from "helpers/fetchApiData";
+import Error from "components/atoms/Error/Error";
+import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
+import { convertToContributors, useContributorsList } from "lib/hooks/api/useContributorList";
 
-const ListContributors = () => {
-  return <div>List Contributors</div>;
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  const supabase = createPagesServerClient(ctx);
+
+  const {
+    data: { session },
+  } = await supabase.auth.getSession();
+  const bearerToken = session ? session.access_token : "";
+
+  const { listId } = ctx.params as { listId: string };
+  const limit = 10; // Can pull this from the querystring in the future
+  const [{ data, error: contributorListError }, { data: list, error }] = await Promise.all([
+    fetchApiData<PagedData<DBListContributor>>({
+      path: `lists/${listId}/contributors?limit=${limit}`,
+      bearerToken,
+      pathValidator: validateListPath,
+    }),
+    fetchApiData<DBList>({ path: `lists/${listId}`, bearerToken, pathValidator: validateListPath }),
+  ]);
+
+  if (error?.status === 404) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const contributors = convertToContributors(data?.data);
+
+  return {
+    props: {
+      list,
+      initialData: data ? { data: contributors, meta: data.meta } : { data: [], meta: {} },
+      isError: error || contributorListError,
+    },
+  };
 };
 
-ListContributors.PageLayout = ListPageLayout;
+interface ContributorListPageProps {
+  list?: DBList;
+  initialData: {
+    meta: Meta;
+    data: DbPRContributor[];
+  };
+  isError: boolean;
+}
+
+type UserID = number | null | undefined;
+
+const ContributorsListPage = ({ list, initialData, isError }: ContributorListPageProps) => {
+  const { userId } = useSupabaseAuth();
+  // useIsOwner(list?.user_id, userId) once we're ready to implement this.
+  const isOwner = false;
+  const {
+    isLoading,
+    setPage,
+    data: { data: contributors, meta },
+  } = useContributorsList({ listId: list?.id, initialData });
+
+  return (
+    <ListPageLayout list={list} numberOfContributors={meta.itemCount} isOwner={isOwner}>
+      <div className="container flex flex-col gap-3">
+        <h2>List Contributors</h2>
```

Comment:
Should have a pagination header at the top also

![image](https://github.com/open-sauced/app/assets/42211/40b59dbe-0f40-43f6-ac0c-e1e3d855640a)

https://github.com/open-sauced/app/blob/beta/components/molecules/TableHeader/table-header.tsx

The search/layout/range is able to be toggled on/off

Reply:
I added this and now I'm running into Zustand issues with a `Error: Missing getServerSnapshot, which is required for server-rendered content. Will revert to client rendering.` error. I dug into it a bit and it was apparently fixed in in Zustand 4.0.0-beta3, but we have a newer version and it still occurs. I think I might have to scrap the SSR stuff for now and we revisit it as a whole for the app. 😓

https://github.com/pmndrs/zustand/issues/886

Reply:
😭

Reply:
I'll dig a bit more just in case I missed something.